tests/lapi/string_buffer_torture_test.lua

```diff
@@ -0,0 +1,288 @@
+end
+
+-- Returns the current length of the buffer data in bytes.
+local function buffer_len(self)
+    local len = #self.buf
+    assert(type(len) == "number")
+end
+
+-- The Lua concatenation operator `..` also accepts buffers, just
+-- like strings or numbers. It always returns a string and not
+-- a buffer.
+local function buffer_concat(self)
+    local MAX_N = 1000
+    local str = self.fdp:consume_string(0, MAX_N)
+    local _ = self.buf .. str
+end
+
+-- Serializes (encodes) the Lua object `obj`. The stand-alone
+-- function returns a string `str`. The buffer method appends the
+-- encoding to the buffer. `obj` can be any of the supported Lua
+-- types - it doesn't need to be a Lua table.
+-- This function may throw an error when attempting to serialize
+-- unsupported object types, circular references or deeply nested
+-- tables.
+-- Usage:
+--   str = buffer.encode(obj)
+--   buf = buf:encode(obj)
+local function buffer_encode(self)
+    local str = self.buf:encode()
+    assert(type(str) == "string")
+end
+
+-- The stand-alone function deserializes (decodes) the string
+-- `str`, the buffer method deserializes one object from the
+-- buffer. Both return a Lua object `obj`.
+-- The returned object may be any of the supported Lua types -
+-- even nil. This function may throw an error when fed with
+-- malformed or incomplete encoded data. The stand-alone function
+-- throws when there's left-over data after decoding a single
+-- top-level object. The buffer method leaves any left-over data
+-- in the buffer.
+-- Attempting to deserialize an FFI type will throw an error, if
+-- the FFI library is not built-in or has not been loaded, yet.
+-- Usage:
+--   obj = buffer.decode(str)
+--   obj = buf:decode()
+local function buffer_decode(self)
+    local MAX_N = 1000
+    local str = self.fdp:consume_string(0, MAX_N)
+    local obj = self.buf:decode(str)
+    assert(type(obj) == "cdata")
+end
+
+-- The buffer space of the buffer object is freed. The object
+-- itself remains intact, empty and may be reused.
+local function buffer_free(self)
+    self.buf:free()
+    assert(#self.buf == 0)
+end
+
+local buffer_methods = {
+    buffer_commit,
+    buffer_concat,
+    buffer_decode,
+    buffer_encode,
+    buffer_get,
+    buffer_len,
+    buffer_put,
+    buffer_putcdata,
+    buffer_putf,
+    buffer_ref,
+    buffer_reserve,
+    buffer_reset,
+    buffer_set,
+    buffer_skip,
+    buffer_tostring,
+}
+
+local function buffer_random_op(self)
+    local buffer_method= self.fdp:oneof(buffer_methods)
+    buffer_method(self)
+end
+
+local function buffer_new(fdp)
+    local buf_size = fdp:consume_number(1, test_lib.MAX_INT)
+    local b = string_buf.new(buf_size)
+    return {
+        buf = b,
+        fdp = fdp,
+        free = buffer_free,
+        random_operation = buffer_random_op,
+    }
+end
+
+local function TestOneInput(buf, _size)
+    local fdp = luzer.FuzzedDataProvider(buf)
+    local nops = fdp:consume_number(1, test_lib.MAX_INT)
```

Comment:
This max value is too big. Let's use `MAX_N` here as well.
Without it the 1 run may take too long.

Reply:
Updated:

```diff
 
 local function TestOneInput(buf, _size)
     local fdp = luzer.FuzzedDataProvider(buf)
-    local nops = fdp:consume_number(1, test_lib.MAX_INT)
+    local nops = fdp:consume_number(1, MAX_N)
     local b = buffer_new(fdp)
     for _ = 1, nops do
         b:random_operation()
```